// @pythnetwork/hermes-client/dist/cjs/zodSchemas.cjs
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: Object.getOwnPropertyDescriptor(all, name).get
    });
}
_export(exports, {
    get api () {
        return api;
    },
    get createApiClient () {
        return createApiClient;
    },
    get schemas () {
        return schemas;
    }
});
const _core = require("@zodios/core");
const _zod = require("zod");
const AssetType = _zod.z.enum([
    "crypto",
    "fx",
    "equity",
    "metal",
    "rates",
    "crypto_redemption_rate",
    "commodities",
    "crypto_index",
    "crypto_nav"
]);
const asset_type = AssetType.nullish();
const RpcPriceIdentifier = _zod.z.string();
const PriceFeedMetadata = _zod.z.object({
    attributes: _zod.z.record(_zod.z.string()),
    id: RpcPriceIdentifier
}).passthrough();
const PriceIdInput = _zod.z.string();
const EncodingType = _zod.z.enum([
    "hex",
    "base64"
]);
const BinaryUpdate = _zod.z.object({
    data: _zod.z.array(_zod.z.string()),
    encoding: EncodingType
}).passthrough();
const RpcPrice = _zod.z.object({
    conf: _zod.z.string(),
    expo: _zod.z.number().int(),
    price: _zod.z.string(),
    publish_time: _zod.z.number().int()
}).passthrough();
const RpcPriceFeedMetadataV2 = _zod.z.object({
    prev_publish_time: _zod.z.number().int().nullable(),
    proof_available_time: _zod.z.number().int().nullable(),
    slot: _zod.z.number().int().gte(0).nullable()
}).partial().passthrough();
const ParsedPriceUpdate = _zod.z.object({
    ema_price: RpcPrice,
    id: RpcPriceIdentifier,
    metadata: RpcPriceFeedMetadataV2,
    price: RpcPrice
}).passthrough();
const PriceUpdate = _zod.z.object({
    binary: BinaryUpdate,
    parsed: _zod.z.array(ParsedPriceUpdate).nullish()
}).passthrough();
const ParsedPublisherStakeCap = _zod.z.object({
    cap: _zod.z.number().int().gte(0),
    publisher: _zod.z.string()
}).passthrough();
const ParsedPublisherStakeCapsUpdate = _zod.z.object({
    publisher_stake_caps: _zod.z.array(ParsedPublisherStakeCap)
}).passthrough();
const LatestPublisherStakeCapsUpdateDataResponse = _zod.z.object({
    binary: BinaryUpdate,
    parsed: _zod.z.array(ParsedPublisherStakeCapsUpdate).nullish()
}).passthrough();
const ParsedPriceFeedTwap = _zod.z.object({
    down_slots_ratio: _zod.z.string(),
    end_timestamp: _zod.z.number().int(),
    id: RpcPriceIdentifier,
    start_timestamp: _zod.z.number().int(),
    twap: RpcPrice
}).passthrough();
const TwapsResponse = _zod.z.object({
    binary: BinaryUpdate,
    parsed: _zod.z.array(ParsedPriceFeedTwap).nullish()
}).passthrough();
const schemas = {
    AssetType,
    asset_type,
    RpcPriceIdentifier,
    PriceFeedMetadata,
    PriceIdInput,
    EncodingType,
    BinaryUpdate,
    RpcPrice,
    RpcPriceFeedMetadataV2,
    ParsedPriceUpdate,
    PriceUpdate,
    ParsedPublisherStakeCap,
    ParsedPublisherStakeCapsUpdate,
    LatestPublisherStakeCapsUpdateDataResponse,
    ParsedPriceFeedTwap,
    TwapsResponse
};
const endpoints = (0, _core.makeApi)([
    {
        method: "get",
        path: "/v2/price_feeds",
        alias: "price_feeds_metadata",
        description: `Get the set of price feeds.

This endpoint fetches all price feeds from the Pyth network. It can be filtered by asset type
and query string.`,
        requestFormat: "json",
        parameters: [
            {
                name: "query",
                type: "Query",
                schema: _zod.z.string().nullish()
            },
            {
                name: "asset_type",
                type: "Query",
                schema: asset_type
            }
        ],
        response: _zod.z.array(PriceFeedMetadata)
    },
    {
        method: "get",
        path: "/v2/updates/price/:publish_time",
        alias: "timestamp_price_updates",
        description: `Get the latest price updates by price feed id.

Given a collection of price feed ids, retrieve the latest Pyth price for each price feed.`,
        requestFormat: "json",
        parameters: [
            {
                name: "publish_time",
                type: "Path",
                schema: _zod.z.number().int()
            },
            {
                name: "ids[]",
                type: "Query",
                schema: _zod.z.array(PriceIdInput)
            },
            {
                name: "encoding",
                type: "Query",
                schema: _zod.z.enum([
                    "hex",
                    "base64"
                ]).optional()
            },
            {
                name: "parsed",
                type: "Query",
                schema: _zod.z.boolean().optional()
            },
            {
                name: "ignore_invalid_price_ids",
                type: "Query",
                schema: _zod.z.boolean().optional()
            }
        ],
        response: PriceUpdate,
        errors: [
            {
                status: 404,
                description: `Price ids not found`,
                schema: _zod.z.void()
            }
        ]
    },
    {
        method: "get",
        path: "/v2/updates/price/latest",
        alias: "latest_price_updates",
        description: `Get the latest price updates by price feed id.

Given a collection of price feed ids, retrieve the latest Pyth price for each price feed.`,
        requestFormat: "json",
        parameters: [
            {
                name: "ids[]",
                type: "Query",
                schema: _zod.z.array(PriceIdInput)
            },
            {
                name: "encoding",
                type: "Query",
                schema: _zod.z.enum([
                    "hex",
                    "base64"
                ]).optional()
            },
            {
                name: "parsed",
                type: "Query",
                schema: _zod.z.boolean().optional()
            },
            {
                name: "ignore_invalid_price_ids",
                type: "Query",
                schema: _zod.z.boolean().optional()
            }
        ],
        response: PriceUpdate,
        errors: [
            {
                status: 404,
                description: `Price ids not found`,
                schema: _zod.z.void()
            }
        ]
    },
    {
        method: "get",
        path: "/v2/updates/price/stream",
        alias: "price_stream_sse_handler",
        description: `SSE route handler for streaming price updates.

The connection will automatically close after 24 hours to prevent resource leaks.
Clients should implement reconnection logic to maintain continuous price updates.`,
        requestFormat: "json",
        parameters: [
            {
                name: "ids[]",
                type: "Query",
                schema: _zod.z.array(PriceIdInput)
            },
            {
                name: "encoding",
                type: "Query",
                schema: _zod.z.enum([
                    "hex",
                    "base64"
                ]).optional()
            },
            {
                name: "parsed",
                type: "Query",
                schema: _zod.z.boolean().optional()
            },
            {
                name: "allow_unordered",
                type: "Query",
                schema: _zod.z.boolean().optional()
            },
            {
                name: "benchmarks_only",
                type: "Query",
                schema: _zod.z.boolean().optional()
            },
            {
                name: "ignore_invalid_price_ids",
                type: "Query",
                schema: _zod.z.boolean().optional()
            }
        ],
        response: PriceUpdate,
        errors: [
            {
                status: 404,
                description: `Price ids not found`,
                schema: _zod.z.void()
            }
        ]
    },
    {
        method: "get",
        path: "/v2/updates/publisher_stake_caps/latest",
        alias: "latest_publisher_stake_caps",
        description: `Get the most recent publisher stake caps update data.`,
        requestFormat: "json",
        parameters: [
            {
                name: "encoding",
                type: "Query",
                schema: _zod.z.enum([
                    "hex",
                    "base64"
                ]).optional()
            },
            {
                name: "parsed",
                type: "Query",
                schema: _zod.z.boolean().optional()
            }
        ],
        response: LatestPublisherStakeCapsUpdateDataResponse
    },
    {
        method: "get",
        path: "/v2/updates/twap/:window_seconds/latest",
        alias: "latest_twaps",
        description: `Get the latest TWAP by price feed id with a custom time window.

Given a collection of price feed ids, retrieve the latest Pyth TWAP price for each price feed.`,
        requestFormat: "json",
        parameters: [
            {
                name: "window_seconds",
                type: "Path",
                schema: _zod.z.number().int().gte(0)
            },
            {
                name: "ids[]",
                type: "Query",
                schema: _zod.z.array(PriceIdInput)
            },
            {
                name: "encoding",
                type: "Query",
                schema: _zod.z.enum([
                    "hex",
                    "base64"
                ]).optional()
            },
            {
                name: "parsed",
                type: "Query",
                schema: _zod.z.boolean().optional()
            },
            {
                name: "ignore_invalid_price_ids",
                type: "Query",
                schema: _zod.z.boolean().optional()
            }
        ],
        response: TwapsResponse,
        errors: [
            {
                status: 404,
                description: `Price ids not found`,
                schema: _zod.z.void()
            }
        ]
    }
]);
const api = new _core.Zodios(endpoints);
function createApiClient(baseUrl, options) {
    return new _core.Zodios(baseUrl, endpoints, options);
}
